src/hotspot/cpu/ppc/assembler_ppc.hpp

```diff
@@ -1424,6 +1515,32 @@ class Assembler : public AbstractAssembler {
   static bool is_lis(int x) {
      return is_addis(x) && inv_ra_field(x)==0;
   }
+  static bool is_paddi_prefix(int x) {
+     return PADDI_PREFIX_OPCODE == (x & PADDI_PREFIX_OPCODE_MASK);
+  }
+  static bool is_paddi_suffix(int x) {
+     return PADDI_SUFFIX_OPCODE == (x & PADDI_SUFFIX_OPCODE_MASK);
+  }
+  static bool is_pli_prefix(int x) {
+     return is_paddi_prefix(x) && inv_r_eo(x) == 0;
+  }
+  static bool is_pli_suffix(int x) {
+     return is_paddi_suffix(x) && inv_ra_field(x) == 0;
+  }
+  static int is_paddi(const int* p, bool is_pli = false) {
+     int32_t* p_inst = (int32_t*)p;
+
+     if (is_aligned(reinterpret_cast<uintptr_t>(p_inst+1), 64) && is_nop(*p_inst)) {
+        p_inst++;  // skip over nop
+     }
+     if (is_paddi_prefix(p_inst[0]) && is_paddi_suffix(p_inst[1])) {
+        return !is_pli ||
+               (inv_r_eo(p_inst[0]) == 0 && inv_ra_field(p_inst[1]) == 0);
+     } else {
+        return false;
+     }
+  }
+  static bool is_pli(const int* p)   { return is_paddi(p, true); }
```

Comment:
is_pli() is defined, but not used, as far as I can tell.

Reply:
It is just intended to show a usage of the second argument of is_paddi(const int* p, bool is_pli = false).  Is it unnecessary, i.e., self-evident?

Reply:
In past open source projects I have worked on, functions that essentially are unreachable in the final product cannot be tested, so there's no way to know if they work as intended. For that reason, only code that is used and/or can be tested in the final product (e.g. via an externally-visible API) would be accepted in a commit.

I don't know if the OpenJDK project follows that rule or not.